### PR TITLE
AJAX mode pick fields 

### PR DESCRIPTION
**Pull request by pglewis**
Fixes: #4388 

- [x] Return full data for the client side models when getting ajax results
- [x] Update the partial collection on the client side after an ajax call
- [x] Test regular autocomplete in ajax mode
- [x] Find out why HTML entities are escaped and not interpreted when the data arrives via ajax

### CODE REVIEW

**Comment by pglewis** [comment]
This one is set for review and testing, all the basics seem to be working now.

**Comment by rob** [comment]
I can confirm that this is working perfectly! Both adding new posts and showing entities. Thanks for the quick fix @pglewis!

**Comment by sc0ttkclark** [comment]
👍

File: classes/fields/pick.php [hunk 1]
```diff
@@ -2079,7 +2079,7 @@ public function get_object_data( $object_params = null ) {
 
 				if ( $autocomplete ) {
 					if ( 0 === $limit ) {
-						$limit = 30;
+						$limit = 3;
```
Comment (sc0ttkclark) [inline classes/fields/pick.php]: 3?

Comment (sc0ttkclark) [inline classes/fields/pick.php]: Nevermind, this ultimately was not changed in the final diff

**Pull request merged**